pkg/ocm/client.go

```diff
@@ -0,0 +1,87 @@
+package ocm
+
+import (
+	"fmt"
+	"time"
+
+	sdkClient "github.com/openshift-online/ocm-sdk-go"
+	"github.com/patrickmn/go-cache"
+)
+
+type Client struct {
+	config     *Config
+	logger     sdkClient.Logger
+	connection *sdkClient.Connection
+
+	Authentication OCMAuthentication
+}
+
+type Config struct {
+	BaseURL      string `envconfig:"OCM_BASE_URL" default:""`
+	ClientID     string `envconfig:"OCM_SERVICE_CLIENT_ID" default:""`
+	ClientSecret string `envconfig:"OCM_SERVICE_CLIENT_SECRET" default:""`
+	SelfToken    string `envconfig:"OCM_SELF_TOKEN" default:""`
+	Debug        bool   `envconfig:"OCM_DEBUG" default:"false"`
+}
+
+func NewClient(config Config) (*Client, error) {
+	// Create a logger that has the debug level enabled:
+	logger, err := sdkClient.NewGoLoggerBuilder().
+		Debug(config.Debug).
+		Build()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to build OCM logger: %s", err.Error())
+	}
+	c := cache.New(1*time.Hour, 30*time.Minute)
+	client := &Client{
+		config: &config,
+		logger: logger,
+	}
+	client.Authentication = &authentication{
```

Comment:
looks strange, 
client variable that has client and param?

Reply:
I agree. see https://github.com/openshift/assisted-service/pull/73#discussion_r467791164

Reply:
this actually the way it's implemented in all OCM examples... but sure, will refactor to improve it. later on :)